client/client.go

```diff
@@ -267,7 +267,7 @@ func (cli *Client) NegotiateAPIVersionPing(p types.Ping) {
 	}
 
 	// if server version is lower than the maximum version supported by the Client, downgrade
-	if versions.LessThan(p.APIVersion, api.DefaultVersion) {
+	if versions.LessThan(p.APIVersion, cli.version) {
```

Comment:
Do we somewhere guard against `cli.version` being > `api.DefaultVersion`? if not, should we take that into account here, and do something like

```go
if versions.LessThan(p.APIVersion, cli.version) && versions.LessThan(p.APIVersion, api.DefaultVersion) {
```